src/osd/PrimaryLogPG.cc

```diff
@@ -2395,6 +2395,14 @@ PrimaryLogPG::cache_result_t PrimaryLogPG::maybe_handle_manifest_detail(
     return cache_result_t::HANDLED_PROXY;
   case object_manifest_t::TYPE_CHUNKED: 
     {
+      // this op only for reading metadata
+      if (op->may_read() && ops.size() == 1) {
+	ceph_osd_op osd_op = ops[0].op;
+	if (osd_op.op == CEPH_OSD_OP_STAT) {
+	    return cache_result_t::NOOP;
+	}
```

Comment:
Do we need to promote even if the op is stat in this case?

Reply:
I don't really want to special case this.  There are actually a bunch of operations that wouldn't require promoting the object and we might even want to avoid it for reads on portions of the object that are actually present.  Let's worry about that later.